### PR TITLE
fix(password): Remove the display of the password in HTML file in forms of service, service template, host and host template

### PR DESCRIPTION
**Pull request by callapa**
## Description

Remove the display of the password in HTML file in forms of service, service template, host and host template.

Fixes # (issue)

The password type input fields of the macros were displayed in plain text in the HTML file.

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [x] 2.8.x
- [x] 19.10.x
- [x] 20.04.x
- [x] 20.10.x
- [x] 21.04.x (master)

## Checklist

- [x] I have followed the **coding style guidelines** provided by Centreon
- [x] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [x] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: www/include/common/common-Func.php [hunk 1]
```diff
@@ -2091,3 +2091,49 @@ function get_child($id_page, $lcaTStr)
...
+ *
+ * @return array<int, int>
+ */
+function getSelectOption()
+{
+    $stringToArray = function ($value) {
+        if (strpos($value, ',') !== false) {
+            return explode(',', $value);
+        }
+        return array($value);
```
Comment (jeremyjaouen) [inline www/include/common/common-Func.php:2106]: need to confirm with @kduret but due to recent changes we can now declare array() as []  on 2.8.x :smiley:

File: www/include/configuration/configObject/host_template_model/formHostTemplateModel.php [hunk 2]
```diff
@@ -43,12 +43,20 @@
...
-        $o = "w";
+        $o = HOST_TEMPLATE_WATCH;
     }
-    $DBRESULT = $pearDB->query("SELECT * FROM host, extended_host_information ehi WHERE host_id = '".$host_id."' AND ehi.host_host_id = host.host_id LIMIT 1");
-
+    $DBRESULT = $pearDB->query(
+        "SELECT * FROM host, extended_host_information ehi
+        WHERE host_id = " . ((int) $host_id)
+        . " AND ehi.host_host_id = host.host_id LIMIT 1"
+    );
```
Comment (jeremyjaouen) [inline www/include/configuration/configObject/host_template_model/formHostTemplateModel.php:59]: Shouldn't you use a prepare execute statement with `?` for this and following queries ?

File: www/include/configuration/configObject/service/formService.php [hunk 3]
```diff
@@ -62,10 +65,12 @@ function myDecodeService($arg)
...
-        $checkres = $pearDB->query("SELECT service_id
-                                        FROM $acldbname.centreon_acl
-                                        WHERE service_id = " . $pearDB->escape($service_id) . "
-                                        AND group_id IN (" . $acl->getAccessGroupsString() . ")");
+        $checkres = $pearDB->query(
+            "SELECT service_id
+            FROM {$acldbname}.centreon_acl
+            WHERE service_id = " . ((int) $service_id)
+            . "  AND group_id IN (" . $acl->getAccessGroupsString() . ")"
+        );
```
Comment (jeremyjaouen) [inline www/include/configuration/configObject/service/formService.php:73]: Same question than https://github.com/centreon/centreon/pull/9298/files#r543139705

File: www/include/configuration/configObject/host_template_model/formHostTemplateModel.php (hunk 2 above)
Comment (callapa) [reply www/include/configuration/configObject/host_template_model/formHostTemplateModel.php:59]: There is no PDO in PHP 5.4

File: www/include/configuration/configObject/service/formService.php (hunk 3 above)
Comment (callapa) [reply www/include/configuration/configObject/service/formService.php:73]: There is no PDO in PHP 5.4

File: www/include/common/common-Func.php (hunk 1 above)
Comment (callapa) [reply www/include/common/common-Func.php:2106]: I keep the same usage throughout the file

File: www/include/configuration/configObject/host_template_model/formHostTemplateModel.php (hunk 2 above)
Comment (kduret) [reply www/include/configuration/configObject/host_template_model/formHostTemplateModel.php:59]: As @jeremyjaouen suggestion, it is possible to use `?` to bind values, which works with PearDB

File: www/include/configuration/configObject/host/formHost.ihtml [hunk 4]
```diff
@@ -455,6 +455,23 @@ jQuery(function() {
 
     });
 
+	/**
+	 * Mechanism to clear the password input field when it gets focus
+	 * and recover its previous value when it lost focus if no new value has been defined.
+	 */
+	jQuery('ul#macro').find('input[id^=\'macroValue_\']').on('click', function () {
+		if (jQuery(this).closest('div').find('input[type=\'checkbox\']:checked').val() === '1') {
```
Comment (sc979) [inline www/include/configuration/configObject/host/formHost.ihtml:463]: odd indentation

File: www/include/configuration/configObject/host_template_model/hostTemplateModel.php [hunk 5]
```diff
@@ -37,17 +37,18 @@
...
-isset($_GET["select"]) ? $cG = $_GET["select"] : $cG = null;
-isset($_POST["select"]) ? $cP = $_POST["select"] : $cP = null;
-$cG ? $select = $cG : $select = $cP;
-
-isset($_GET["dupNbr"]) ? $cG = $_GET["dupNbr"] : $cG = null;
-isset($_POST["dupNbr"]) ? $cP = $_POST["dupNbr"] : $cP = null;
-$cG ? $dupNbr = $cG : $dupNbr = $cP;
+$host_id = filter_var(
+    call_user_func(function () {
+        if (isset($_GET["host_id"])) {
```
Comment (sc979) [inline www/include/configuration/configObject/host_template_model/hostTemplateModel.php:42]: $host_id -> $hostId

what would happen if $GET['host_id'] is a string, (which will return $host_id = false) ?

File: www/include/configuration/configObject/host_template_model/hostTemplateModel.php [hunk 6]
```diff
@@ -62,6 +63,15 @@
...
 
+$select = filter_var_array(
+    getSelectOption(),
+    FILTER_VALIDATE_INT
+);
+$dupNbr = filter_var_array(
+    getDuplicateNumberOption(),
+    FILTER_VALIDATE_INT
+);
+
```
Comment (sc979) [inline www/include/configuration/configObject/host_template_model/hostTemplateModel.php:74]: IMHO : you shouldn't process $select or $dupNbr or $host_id = false

**Review by sc979 — COMMENTED** [review]
Kindly update the licence

File: www/include/configuration/configObject/host_template_model/hostTemplateModel.php (hunk 5 above)
Comment (callapa) [reply www/include/configuration/configObject/host_template_model/hostTemplateModel.php:42]: Read the function 'getDuplicateNumberOption'

**Review by sc979 — APPROVED** [review]

**Review by jeremyjaouen — APPROVED** [review]

**Review by kduret — APPROVED** [review]

**Pull request merged**